Week9 - homework SQL/LibrariesManagement.sql

```diff
@@ -0,0 +1,298 @@
+VALUES(114, 1)
+INSERT INTO Sales
+VALUES(98, 4)
+INSERT INTO Sales
+VALUES(321, 3)
+INSERT INTO Sales
+VALUES(987, 2)
+INSERT INTO Sales
+VALUES(89, 7)
+INSERT INTO Sales
+VALUES(56, 8)
+INSERT INTO Sales
+VALUES(347, 5)
+INSERT INTO Sales
+VALUES(563, 11)
+INSERT INTO Sales
+VALUES(97, 6)
+INSERT INTO Sales
+VALUES(224, 9)
+INSERT INTO Sales
+VALUES(90, 12)
+INSERT INTO Sales
+VALUES(542, 10)
+
+SELECT *
+FROM Sales;
+
+--exercise2
+--Display all the librarians together with the library name they work in ordered by librarian name and library name descending.
+SELECT L.Name, Lib.Name
+FROM Librarians L
+INNER JOIN Libraries Lib ON Lib.Id = L.LibraryId
+ORDER BY L.Name DESC;
+
+SELECT L.Name, Lib.Name
+FROM Librarians L
+INNER JOIN Libraries Lib ON Lib.Id = L.LibraryId
+ORDER BY Lib.Name DESC;
+
+--exercise3
+--Update all the street names to start with "STR.".
+UPDATE Libraries
+SET StreetName = 'STR. ' + Streetname;   --I didn't know how to undo the 1st update. That's why the 1st try of 'str' is still there
+
+--exercise4
+--Display the authors' email for those who have more than 3 books published.
+SELECT AuthorEmail
+FROM Books
+GROUP BY AuthorEmail
+HAVING COUNT(AuthorLastName) > 3;
+
+--exercise5
+--Display the library name and the book title of the largest sale (the highest number of copies sold in one sale).
+SELECT MAX(S.NumberOfCopies), Lib.Name, B.Title    --I'm so stuck...
+FROM Sales S
+JOIN Books_Libraries BL ON S.BookLibraryId = BL.Id
+JOIN Libraries Lib ON Lib.Id = BL.LibraryId
+JOIN Books B ON B.Id = BL.BookId;
+
+--exercise6
+--Display all the librarians that are not on holiday and are hired more than 5 years ago.
+SELECT Name 
+FROM Librarians
+WHERE IsOnHoliday = 0 AND HireDate < GETDATE() - YEAR(5)
+
+--exercise7
+--Create a stored procedure that displays, based on the book title, the number of libraries in which the book is found. Call the procedure with three different book titles.
+GO
+CREATE PROCEDURE CountLibraries(
+    @BookTitle VARCHAR(50)
+)
+AS
+BEGIN
+SELECT COUNT(LibraryId) AS NumberOfLibraries
+FROM Books_Libraries
+JOIN Books ON Books.Id = Books_Libraries.BookId
+WHERE @BookTitle = Books.Title
+END
+GO
+
+exec
+CountLibraries'Prietena mea geniala';
+exec
+CountLibraries'Nascut in afara legii';
+exec
+CountLibraries'Ciuleandra';
+
+--exercise8
+--Create a function that returns, based on the author's email, the number of copies sold for each of his / her books. Call the function with three different emails.
+--I messed up here -_-
+GO
+CREATE FUNCTION dbo.CopiesSoldByAuthorEmail(
+    @Email VARCHAR(50)
+)
+RETURNS INT
+AS
+BEGIN
+DECLARE @numberOfCopies INT;
+SET @numberOfCopies = 
+    (SELECT COUNT(Sales.NumberOfCopies)
+    FROM Sales    
+    JOIN Books_Libraries ON Sales.BookLibraryId = Books_Libraries.Id
+    JOIN Books ON Books.Id = Books_Libraries.BookId
+    WHERE Books.AuthorEmail = @Email);
+RETURN @numberOfCopies;
+END;
+GO
+
+select dbo.CopiesSoldByAuthorEmail('ferrante@gmail.com');
```

Comment:
The code is running with no error which is good. :) It doesn't seem to do the right thing though. Please have a look on my example once it's published.